src/types/target.ts

```diff
@@ -254,4 +254,32 @@ export class Target implements TargetJson {
 
     return json
   }
+
+  toJsonNoDefaults(): TargetJson {
+    const json: TargetJson = {
+      name: this.name,
+      serverUrl: this.serverUrl,
+      serverType: this.serverType,
+      allowInsecureRequests: this.allowInsecureRequests,
+      appLoc: this.appLoc,
+      macroFolders: this.macroFolders,
+      programFolders: this.programFolders,
+      docConfig: this.docConfig,
+      authConfig: this.authConfig,
+      buildConfig: this.buildConfig,
+      jobConfig: this.jobConfig,
+      serviceConfig: this.serviceConfig,
+      streamConfig: this.streamConfig,
```

Comment:
Won't this cause `undefined` to be returned when a property has not been set?

Reply:
if a property is undefined in an object, means it doesn't exist.

And that's exactly we want, when updating a target, it shouldn't add additional attributes to current.

Reply:
Yeah, but if for example `this.streamConfig` is undefined, then this function will return an object containing `streamConfig: undefined`. Is that what we want?

Reply:
Also, do we really need this to be a separate function, or can we just modify the original one to not return the defaults?

Reply:
Removed new function, added checks in current `toJson` and passed boolean `withDefaults`.